modules/post/windows/gather/enum_computers.rb

```diff
@@ -31,74 +36,87 @@ def initialize(info = {})
     )
   end
 
-  # Run Method for when run command is issued
   def run
-    print_status("Running module against #{sysinfo['Computer']}") if !sysinfo.nil?
-    domain = get_domain
-    if !domain.empty?
-      hostname_list = get_domain_computers
-      list_computers(domain, hostname_list)
+    hostname = sysinfo.nil? ? cmd_exec('hostname') : sysinfo['Computer']
+    print_status("Running module against #{hostname} (#{session.session_host})")
+
+    domain = get_domain_name
+
+    fail_with(Failure::Unknown, 'Could not retrieve domain name. Is the host part of a domain?') unless domain
+
+    netbios_domain_name = domain.split('.').first.upcase
+
+    hostname_list = get_domain_computers
+
+    if hostname_list.empty?
+      print_error('No computers found')
+      return
     end
+
+    list_computers(netbios_domain_name, hostname_list)
   end
 
-  def gethost(hostname)
-    ## get IP for host
-    vprint_status("Looking up IP for #{hostname}")
-    result = client.net.resolve.resolve_host(hostname)
+  # Takes the host name and makes use of nslookup to resolve the IP
+  #
+  # @param [String] host Hostname
+  # @return [String] ip The resolved IP
+  def resolve_host(host)
+    vprint_status("Looking up IP for #{host}")
+    return host if Rex::Socket.dotted_ip?(host)
 
-    return nil if result[:ip].nil? || result[:ip].blank?
+    ip = []
+    data = cmd_exec("nslookup #{host}")
+    if data =~ /Name/
+      # Remove unnecessary data and get the section with the addresses
+      returned_data = data.split(/Name:/)[1]
+      # check each element of the array to see if they are IP
+      returned_data.gsub(/\r\n\t |\r\n|Aliases:|Addresses:|Address:/, ' ').split(' ').each do |e|
+        if Rex::Socket.dotted_ip?(e)
+          ip << e
+        end
+      end
+    end
 
-    return result[:ip]
+    if ip.blank?
+      'Not resolvable'
+    else
+      ip.join(', ')
+    end
   end
 
-  # List Members of a domain group
   def get_domain_computers
     computer_list = []
     devisor = "-------------------------------------------------------------------------------\r\n"
     raw_list = cmd_exec('net view').split(devisor)[1]
```

Comment:
> Suggestion (won't let me make a github suggestion across deleted lines):
> 
> ```
>     raw_list = cmd_exec('net view')
>     return [] unless raw_list.include?('The command completed successfully')
>     raw_list = raw_list.split(devisor)[1]
> ```
> 
> or if you wanted to print the system error returned by windows 🤷
> 
> ```
>   net_view_response = cmd_exec('net view')
>   if net_view_response.include?(/System error/)
>     print_error("The net view command failed with: #{net_view_response}")
>     return []
>   end
>   raw_list = net_view_response.split(devisor)[1]
> ```

This error handling will fail on non-English systems.

Reply:
Correct @bcoles, thanks for the reminder there. The devisor will never be present if Windows returns an error, we should maybe be checking for that instead of the english words success / error.

Reply:
Thanks for the suggestion. I have updated it to now print an error if the devisor isn't present and to return a blank array.